File: Code - Python/Algorithm - CST 3D/cosspace_half.py
import numpy as np

def cosspace_half(startP,endP,N=100):

    # Versão modificada da funçãoo cosspace. Gera apenas metade da distribuição.
    x = np.zeros(N); x[N-1] = endP
    angleInc = np.pi/(N-1)/2

    curAngle = angleInc
    for i in range(1,N-1):
        x[i]=endP*(1-np.cos(curAngle))
        curAngle += angleInc

    return x

File: Code - Octave/Geometry/cosspace_half.m
function x = cosspace_half(startP,endP,N)
    
% Versão modificada da função cosspace. Gera apenas metade da distribuição.


x = zeros(1,N); x(N) = endP;
angleInc = pi/(N-1)/2;
    
curAngle=angleInc;
for i = 2:N-1
    x(i)=endP*(1-cos(curAngle));
    curAngle=curAngle+angleInc;
end
     
end